src/test/fuzz/prevector.cpp

```diff
@@ -0,0 +1,268 @@
+// Copyright (c) 2015-2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+
+#include <vector>
+#include <prevector.h>
+
+#include <reverse_iterator.h>
+#include <serialize.h>
+#include <streams.h>
+
+namespace {
+
+template<unsigned int N, typename T>
+class prevector_tester {
+    typedef std::vector<T> realtype;
+    realtype real_vector;
+    realtype real_vector_alt;
+
+    typedef prevector<N, T> pretype;
+    pretype pre_vector;
+    pretype pre_vector_alt;
+
+    typedef typename pretype::size_type Size;
+
+public:
+    void test() {
+        const pretype& const_pre_vector = pre_vector;
+        assert(real_vector.size() == pre_vector.size());
+        assert(real_vector.empty() == pre_vector.empty());
+        for (Size s = 0; s < real_vector.size(); s++) {
+             assert(real_vector[s] == pre_vector[s]);
+             assert(&(pre_vector[s]) == &(pre_vector.begin()[s]));
+             assert(&(pre_vector[s]) == &*(pre_vector.begin() + s));
+             assert(&(pre_vector[s]) == &*((pre_vector.end() + s) - real_vector.size()));
+        }
+        // assert(realtype(pre_vector) == real_vector);
+        assert(pretype(real_vector.begin(), real_vector.end()) == pre_vector);
+        assert(pretype(pre_vector.begin(), pre_vector.end()) == pre_vector);
+        size_t pos = 0;
+        for (const T& v : pre_vector) {
+             assert(v == real_vector[pos++]);
+        }
+        for (const T& v : reverse_iterate(pre_vector)) {
+             assert(v == real_vector[--pos]);
+        }
+        for (const T& v : const_pre_vector) {
+             assert(v == real_vector[pos++]);
+        }
+        for (const T& v : reverse_iterate(const_pre_vector)) {
+             assert(v == real_vector[--pos]);
```

Comment:
I think the assert-side-effects linter complains about this, so you might have two options:

* Disable the linter for the fuzz tests
* Remove the side effect

```
Assertions should not have side effects:

src/test/fuzz/prevector.cpp:             assert(v == real_vector[pos++]);

src/test/fuzz/prevector.cpp:             assert(v == real_vector[--pos]);

src/test/fuzz/prevector.cpp:             assert(v == real_vector[pos++]);

src/test/fuzz/prevector.cpp:             assert(v == real_vector[--pos]);

^---- failure generated from test/lint/lint-assertions.sh

Reply:
Fixed by removing the side effect.